pywatts/core/pipeline.py

```diff
@@ -219,33 +219,31 @@ def _run(self, data: Union[pd.DataFrame, xr.Dataset], mode: ComputationMode, sum
 
         if isinstance(data, pd.DataFrame):
             data = data.to_xarray()
-        elif isinstance(data, dict):
+            data = {key: data[key] for key in data.data_vars}
+        elif isinstance(data, xr.Dataset):
+            data = {key: data[key] for key in data.data_vars}
+
+        if isinstance(data,dict):
             for key in data:
                 if not isinstance(data[key], xr.DataArray):
                     raise WrongParameterException(
                         "Input Dict does not contain xr.DataArray objects.",
                         "Make sure to pass Dict[str, xr.DataArray].",
                         self.name
                     )
-            data = xr.Dataset(data)
-
-        if isinstance(data, xr.Dataset):
             if self.current_run_setting.online_start is not None:
                 # First only _transform should be called (no summary, no online) on the data before online_start.
                 # Afterwards, comp is called (_transform and summaries using online simulation)
                 index_name = _get_time_indexes(data)[0]
                 self._transform({key: data[key].sel(
-                    **{index_name: data[key][index_name] < self.current_run_setting.online_start}) for key in
-                    data.data_vars}, False)
+                    **{index_name: data[key][index_name] < self.current_run_setting.online_start }) for key in data}, False)
                 for step in self.id_to_step.values():
                     step.reset(keep_buffer=True)
                     step.set_run_setting(self.current_run_setting.clone())
-                return self._comp({key: data[key].sel(
-                    **{index_name: data[key][index_name] >= self.current_run_setting.online_start}) for key in
-                    data.data_vars}
-                    , summary_formatter, self.batch, start=self.current_run_setting.online_start)
+                return self._comp({key: data[key].sel(**{index_name: data[key][index_name] >= self.current_run_setting.online_start}) for key in data},
+                                  summary_formatter, self.batch, start=self.current_run_setting.online_start)
             else:
-                return self._comp({key: data[key] for key in data.data_vars}, summary_formatter, self.batch)
+                return self._comp(data, summary_formatter, self.batch)
 
         raise WrongParameterException(
```

Comment:
If we use 'elif' we could use 'else' as default if no type matches because currently, it looks a little bit odd for me if a raise of an exception is at the end of a method without any constraints.